### PR TITLE
Add npm integration test for yarn registry

### PR DESCRIPTION
**Pull request by ejegrova**
STONEBLD-1420

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- n/a Code coverage from testing does not decrease and new code is covered
- n/a Docs updated (if applicable)
- n/a Docs links in the code are still valid (if docs were updated)

### CODE REVIEW

File: tests/integration/test_data/npm_lockfile3_yarn_registry/bom.json [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+      "purl": "pkg:npm/array-flatten@1.1.1",
+      "type": "library",
+      "version": "1.1.1"
+    },
+    {
+      "name": "bar",
+      "purl": "pkg:npm/bar@1.0.0?vcs_url=git%2Bhttps://github.com/cachito-testing/npm-cachi2-registry-yarnpkg.git%40d99077c1388bd575dc07d287b8af729da1557d1c#bar",
+      "type": "library",
+      "version": "1.0.0"
+    },
```
Comment (chmeliik) [inline tests/integration/test_data/npm_lockfile3_yarn_registry/bom.json:51]: Do we want the workspaces and their extra dependencies here? IMO there's value in keeping the different-lockfileVersion tests as similar as possible

**Comment by chmeliik** [comment]
from the lockfile-v3 branch: https://github.com/cachito-testing/npm-cachi2-registry-yarnpkg/blob/d99077c1388bd575dc07d287b8af729da1557d1c/package-lock.json#L57-L61

```json
    "node_modules/@types/prop-types": {
      "version": "15.7.5",
      "resolved": "https://registry.yarnpkg.com/@types/prop-types/-/prop-types-15.7.5.tgz",
      "integrity": "sha1-XxnSuFqY6VWANvajysyIGUIPBc8= sha512-JCB8C6SnDoQf0cNycqd/35A7MjcnK+ZTqE7judS6o7utxUCg6imJg3QK2qzHKszlTjcj2cn+NwMB2i96ubpj7w=="
    },
```

I was almost sure that this would break cachi2, but we successfully verify the sha1 checksum

```
2023-06-30 17:19:03,039 DEBUG types-prop-types-15.7.5.tgz: sha1 checksum matches: 5f19d2b85a98e9558036f6a3cacc8819420f05cf
```

Turns out that we accidentally parse the only the `sha1-XxnSuFqY6VWANvajysyIGUIPBc8=` part because `b64decode` is smart like that

```python
In [6]: base64.b64decode("XxnSuFqY6VWANvajysyIGUIPBc8=").hex()
Out[6]: '5f19d2b85a98e9558036f6a3cacc8819420f05cf'

In [7]: base64.b64decode("XxnSuFqY6VWANvajysyIGUIPBc8= and some other nonsense").hex()
Out[7]: '5f19d2b85a98e9558036f6a3cacc8819420f05cf'
```

But we shouln't be using a cryptographically broken checksum algorithm when a stronger one is available. Let's file a story to handle this similarly to how cachito yarn does it https://github.com/containerbuildsystem/cachito/blob/e9e68339d4a7d422e70dfd45af44403a921b8b61/cachito/workers/pkg_managers/yarn.py#L287

File: tests/integration/test_data/npm_lockfile3_yarn_registry/bom.json [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+    {
+      "name": "negotiator",
+      "purl": "pkg:npm/negotiator@0.6.3",
+      "type": "library",
+      "version": "0.6.3"
+    },
+    {
+      "name": "npm_test",
+      "purl": "pkg:npm/npm_test@1.1.0?vcs_url=git%2Bhttps://github.com/cachito-testing/npm-cachi2-registry-yarnpkg.git%40f2daad2b7a844779bdae8b3b66e6ec1c9de70beb",
```
Comment (chmeliik) [inline tests/integration/test_data/npm_lockfile3_yarn_registry/bom.json:276]: nitpick: the name of the main package is different between lockfile v1 and lockfile v3

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Comment by chmeliik** [comment]
Do we want to explain in the commit message what we're testing here?

**Review by taylormadore — APPROVED** [review]
LGTM :+1:

**Review by lkolacek — APPROVED** [review]
LGTM :+1:

**Comment by ejegrova** [comment]
> Do we want to explain in the commit message what we're testing here?

Updated, although I'm not sure if that's what you want. It seems pretty self-explanatory to me.

**Comment by chmeliik** [comment]
> > Do we want to explain in the commit message what we're testing here?
> 
> Updated, although I'm not sure if that's what you want. It seems pretty self-explanatory to me.

If I didn't already know, I might be curious about

* what it means to "use the yarn registry for npm dependencies" (resolved url in package-lock.json uses registry.yarnpkg.com instead of registry.npmjs.org)
* when such a situation would ever occur (converting a yarn.lock to package-lock.json)
* what is the difference between the registries (none, it's just two different CNAMEs for one server)
* how cachi2 is expected to handle them (exactly the same as registry.npmjs.org)

**Comment by ejegrova** [comment]
> > > Do we want to explain in the commit message what we're testing here?
> > 
> > 
> > Updated, although I'm not sure if that's what you want. It seems pretty self-explanatory to me.
> 
> If I didn't already know, I might be curious about
> 
>     * what it means to "use the yarn registry for npm dependencies" (resolved url in package-lock.json uses registry.yarnpkg.com instead of registry.npmjs.org)
> 
>     * when such a situation would ever occur (converting a yarn.lock to package-lock.json)
> 
>     * what is the difference between the registries (none, it's just two different CNAMEs for one server)
> 
>     * how cachi2 is expected to handle them (exactly the same as registry.npmjs.org)

Thanks, done

**Pull request merged**